EIPS/eip-5450.md

```diff
@@ -29,150 +29,40 @@ In particular, this extended code validation eliminates the need for EVM stack u
-
-Given the new deploy-time guarantee of no function underflowing its stack frame, `JUMPF` instruction can check only that there is enough items for input arguments of the callee on the stack, without making sure there is exactly `inputs` items and not more. 
-
-Therefore, the previously defined behavior of discarding extra items is removed, and only the check for enough inputs is done:
-
-> 2. If data stack has less than `caller_stack_height + type[code_section_index].inputs` items, execution results in exceptional halt.
-
-With this change `JUMPF` operation complexity does not depend on `ouputs` value and is constant-time, therefore the price of `JUMPF` is lowered to 3 gas.
-
 ## Rationale
 
 ### Stack overflow check only in CALLF
 
-In the current proposal stack overflow checks are unchanged (i.e. are done for every instruction). However, we can provide more efficient variant where stack overflow check is performed only in `CALLF` instruction and uses called function's `max_stack_height` information for this. This decreases flexibility of an EVM program because `max_stack_height` corresponds to the worst-case control-flow path in the function.
-Moreover, the `max_stack_height` computed during validation must be stored alongside the code. This can be in the EOF itself or in implementation-defined format.
-
-### Unreachable code
-
-The current validation algorithm ignores *unreachable* instructions. The algorithm can be extended to reject any code having any unreachable instructions but additional instructions traversal is needed (or more efficient algorithm must be developed).
-
-### Clean stack upon termination
-
-It is currently required that the EVM stack is empty (in the current function context) after any terminating instruction (`STOP`, `RETURN`, etc, but also `RETF`). This requirement can be lifted.
-
-This can be used for implementing more efficient early exits from a function (e.g. assertion failure).
-
-For "exit" instructions which terminates the whole program execution (`STOP`, `RETURN`, etc) this is no change comparing to pre-EOF EVM. I.e. some *garbage* can be left on the stack. Cleaning the stack does not improve EVM implementation performance but makes the EVM programs potentially cost more (compiler is required to insert additional `POP` instructions).
-
-For `RETF` semantic would be more complicated. For `n` function outputs and `s` the stack height at `RETF` the EVM must erase `s-n` non-top stack items and move the `n` stack items to the place of erased ones. Cost of such operation may be relatively cheap but is not constant.
+In this EIP, we provide a more efficient variant of the EVM where stack overflow check is performed only in `CALLF` instruction using the called function's `max_stack_height` information. This decreases flexibility of an EVM program because `max_stack_height` corresponds to the worst-case control-flow path in the function.
```

Comment:
This is not fully done. E.g. the `CALLF` would need to check `max_stack_height` at runtime.

Reply:
Does something need to change here? Or you're referring to the outstanding PR to 5450 that defines this behavior?

Reply:
How `CALLF` operates, but I will fix it later.